### PR TITLE
NDRCN 124 Changes

### PR DESCRIPTION
**Pull request by somaraja**

### CODE REVIEW

File: app/controllers/AmendCaseSendInformationController.scala [hunk 1]
```diff
@@ -88,7 +88,7 @@ class AmendCaseSendInformationController @Inject()(
     renderFileVerificationStatus(reference, request.userAnswers.fileUploadState)
   }
 
-  final def removeFileUploadByReference(reference: String): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+  final def removeFileUploadByReference(reference: String, mode : Mode = NormalMode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
```
Comment (lmugnai) [inline app/controllers/AmendCaseSendInformationController.scala:91]: @somaraja we've added this parameter but it's not used. Do we need it?

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 2]
```diff
@@ -18,10 +18,11 @@ package controllers
 
 import controllers.actions._
 import forms.ClaimRepaymentTypeFormProvider
+
 import javax.inject.Inject
-import models.Mode
+import models.{ClaimRepaymentType, Mode}
 import navigation.Navigator
-import pages.ClaimRepaymentTypePage
+import pages.{ClaimRepaymentTypePage, CustomsDutyDueToHMRCPage, CustomsDutyPaidPage, OtherDutiesDueToHMRCPage, OtherDutiesPaidPage, VATDueToHMRCPage, VATPaidPage}
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:25]:
```suggestion
import pages._
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 3]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
         value =>
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(ClaimRepaymentTypePage, value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(ClaimRepaymentTypePage, mode, updatedAnswers))
+            removeCustomsDutyDue <-
+              Future.fromTry(updatedAnswers.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Customs) match {
+                case false =>
+                  updatedAnswers.remove(CustomsDutyDueToHMRCPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:76]:
```suggestion
                case true =>
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 4]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
+                  updatedAnswers.set(ClaimRepaymentTypePage, value)
+              })
+            removeCustomsPaid <-
+              Future.fromTry(removeCustomsDutyDue.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Customs) match {
+                case false =>
+                  removeCustomsDutyDue.remove(CustomsDutyPaidPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:83]:
```suggestion
                case true =>
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 5]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
+                  removeCustomsDutyDue.set(ClaimRepaymentTypePage, value)
+              })
+            removeVATDue <-
+              Future.fromTry(removeCustomsPaid.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Vat) match {
+                case false =>
+                  removeCustomsPaid.remove(VATDueToHMRCPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:90]:
```suggestion
                case true =>
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 6]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
+                  removeCustomsPaid.set(ClaimRepaymentTypePage, value)
+              })
+            removeVATPaid <-
+              Future.fromTry(removeVATDue.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Vat) match {
+                case false =>
+                  removeVATDue.remove(VATPaidPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:97]:
```suggestion
                case true =>
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 7]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
+                  removeVATDue.set(ClaimRepaymentTypePage, value)
+              })
+            removeOtherDutiesDue <-
+              Future.fromTry(removeVATPaid.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Other) match {
+                case false =>
+                  removeVATPaid.remove(OtherDutiesDueToHMRCPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:104]:
```suggestion
                case true =>
```

File: app/controllers/ClaimRepaymentTypeController.scala [hunk 8]
```diff
@@ -61,16 +62,57 @@ class ClaimRepaymentTypeController @Inject()(
...
+                  removeVATPaid.set(ClaimRepaymentTypePage, value)
+              })
+            removeOtherDutiesPaidDue <-
+              Future.fromTry(removeOtherDutiesDue.get(ClaimRepaymentTypePage).get.contains(ClaimRepaymentType.Other) match {
+                case false =>
+                  removeOtherDutiesDue.remove(OtherDutiesPaidPage)
+                case _ =>
```
Comment (lmugnai) [inline app/controllers/ClaimRepaymentTypeController.scala:111]:
```suggestion
                case true =>
```

File: app/controllers/CustomsRegulationTypeController.scala [hunk 9]
```diff
@@ -18,10 +18,12 @@ package controllers
...
 import controllers.actions._
 import forms.CustomsRegulationTypeFormProvider
+import models.CustomsRegulationType.UKCustomsCodeRegulation
+
 import javax.inject.Inject
-import models.{Mode, NumberOfEntriesType, UserAnswers}
+import models.{CustomsRegulationType, Mode, NumberOfEntriesType, UserAnswers}
 import navigation.Navigator
-import pages.{CustomsRegulationTypePage, NumberOfEntriesTypePage}
+import pages.{ArticleTypePage, CustomsRegulationTypePage, NumberOfEntriesTypePage, UkRegulationTypePage}
```
Comment (lmugnai) [inline app/controllers/CustomsRegulationTypeController.scala:26]: Why have these been added? I see no new reference to them.

File: conf/messages.en [hunk 10]
```diff
@@ -660,3 +682,8 @@ amend-file-upload.help.fileFormat2= .jpg
 amend-file-upload.help.fileFormat3= .pdf
 amend-file-upload.help.fileFormat4= .png
 amend-file-upload.help.fileFormat5= .xls
+
+your.details.checkYourAnswersLabel = Your details
+contact.details.checkYourAnswersLabel = Contact details
+payment.information.checkYourAnswersLabel = Payment information
+importer.details.checkYourAnswersLabel = Importer details
```
Comment (lmugnai) [inline conf/messages.en:689]:
```suggestion
importer.details.checkYourAnswersLabel = Importer details

```

**Review by chandaksweta — CHANGES_REQUESTED** [review]
Can the commits be sqashed..?

File: app/controllers/AmendCaseSendInformationController.scala (hunk 1 above)
Comment (somaraja) [reply app/controllers/AmendCaseSendInformationController.scala:91]: @lmugnai we need this for CYA page.

File: app/controllers/ClaimRepaymentTypeController.scala (hunk 3 above)
Comment (somaraja) [reply app/controllers/ClaimRepaymentTypeController.scala]: @lmugnai this is fixed

File: app/controllers/ClaimRepaymentTypeController.scala (hunk 4 above)
Comment (somaraja) [reply app/controllers/ClaimRepaymentTypeController.scala]: @lmugnai this is fixed

File: app/controllers/ClaimRepaymentTypeController.scala (hunk 6 above)
Comment (somaraja) [reply app/controllers/ClaimRepaymentTypeController.scala]: @lmugnai this is fixed

File: app/controllers/ClaimRepaymentTypeController.scala (hunk 7 above)
Comment (somaraja) [reply app/controllers/ClaimRepaymentTypeController.scala]: @lmugnai this is fixed

File: app/controllers/ClaimRepaymentTypeController.scala (hunk 8 above)
Comment (somaraja) [reply app/controllers/ClaimRepaymentTypeController.scala]: @lmugnai this is fixed

File: app/controllers/CustomsRegulationTypeController.scala (hunk 9 above)
Comment (somaraja) [reply app/controllers/CustomsRegulationTypeController.scala]: @lmugnai this is fixed

File: conf/messages.en (hunk 10 above)
Comment (somaraja) [reply conf/messages.en:689]: @lmugnai not sure what needs to be fixed here

**Pull request merged**